python/essex-config/essex_config/utils.py

```diff
@@ -0,0 +1,16 @@
+"""Utility functions for the essex_config package."""
+
+from string import Template
+from typing import Any
+
+
+def parse_string_template(value: Any, data: dict[str, Any]) -> Any:
+    """Parse the string templates in str, list[str], and dict[, str]."""
+    if isinstance(value, str):
+        return Template(value).substitute(data)
+    if isinstance(value, list):
+        return [parse_string_template(item, data) for item in value]
+    if isinstance(value, dict):
+        return {key: parse_string_template(val, data) for key, val in value.items()}
```

Comment:
could we use match here? Something like this:

```python
value_type = type(value)

match value_type:
    case str:
        return Template(value).substitute(data)
    case list:
        return [...]
    case dict:
        return {...}
    case _:
        return value

Reply:
I had to switch on `value_type.__name__` instead of `value_type` otherwise I get an error about unreachable code and about shadowing builtin variables. 

![image](https://github.com/user-attachments/assets/1a6b7433-b2cb-4eeb-9733-ac1a06db9b5d)

Here is the modified code that works:

```python
value_type: type = type(value)
match value_type.__name__:
    case "str":
        return Template(value).substitute(data)
    case "list":
        return [parse_string_template(item, data) for item in value]
    case "dict":
        return {key: parse_string_template(val, data) for key, val in value.items()}
    case _:
        return value
```

Reply:
Can you try:
```python
match value:
    case str():
       ...
    case list():
       ...
```

https://stackoverflow.com/questions/72295812/python-match-case-by-type-of-value